pkgs/development/python-modules/rdkit/default.nix

```diff
@@ -0,0 +1,141 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, fetchzip
+, cmake
+, boost
+, catch
+, inchi
+, cairo
+, eigen
+, python
+, rapidjson
+, maeparser
+, coordgenlibs
+, numpy
+, stdenv
+, git
+}:
+
+let
+  external = {
+    avalon = fetchzip {
+      url = "http://sourceforge.net/projects/avalontoolkit/files/AvalonToolkit_1.2/AvalonToolkit_1.2.0.source.tar";
+      sha256 = "0nhxfxckb5a5qs0g148f55yarhncqjgjzcvdskkv9rxi2nrs7160";
+      stripRoot = false;
+    };
+    yaehmop = fetchFromGitHub {
+      owner = "greglandrum";
+      repo = "yaehmop";
+      rev = "1b13b52e2738a77715b1bad876e3b4e93f2b5269";
+      sha256 = "1jp7wz8win4mgwxkaz2gnrgsaaqgln04n2lwgfr96isdv1klf62d";
+    };
+    freesasa = fetchFromGitHub
+      {
+        owner = "mittinatten";
+        repo = "freesasa";
+        rev = "2.0.3";
+        sha256 = "0x686zm9fpyg5647fdgxnxgbwav99nc6ymh4bmkr2063yyda4kzc";
+      };
+  };
+in
+buildPythonPackage rec {
+  version = "2020.09.3";
+  pname = "rdkit";
+
+  src = let
+    versionTag = stdenv.lib.replaceStrings ["."] ["_"] version;
+  in
+    fetchFromGitHub {
+      owner = pname;
+      repo = pname;
+      rev = "Release_${versionTag}";
+      sha256 = "1k1wvzcd1yfx4nhz3iq2aaadzdk5w6sfcb4imhvm7pkbzij0nicx";
+  };
+
+  unpackPhase = ''
+    mkdir -p source/External/AvalonTools/avalon source/External/YAeHMOP/yaehmop source/External/FreeSASA/freesasa
+    cp -r ${src}/* source
+    cp -r ${external.avalon}/SourceDistribution/* source/External/AvalonTools/avalon
+    cp -r ${external.yaehmop}/* source/External/YAeHMOP/yaehmop
+    cp -r ${external.freesasa}/* source/External/FreeSASA/freesasa
```

Comment:
all of these seem to have ebeen vendored: https://github.com/rdkit/rdkit/tree/master/External do we need to check them out?

Reply:
They're not vendored. The CMake scripts automatically download them from these locations if left to run normally, which doesn't work on nix.

Reply:
really curious why they didn't just use submodules on most of these.

`git submodule update --init` seems easier than re-implementing it through bash, then calling it through CMake